nifi-nar-bundles/nifi-solr-bundle/nifi-solr-processors/src/main/java/org/apache/nifi/processors/solr/GetSolr.java

```diff
@@ -170,159 +210,213 @@ protected void init(final ProcessorInitializationContext context) {
-            final StopWatch stopWatch = new StopWatch(true);
-            QueryResponse response = req.process(getSolrClient());
-            stopWatch.stop();
-
-            long duration = stopWatch.getDuration(TimeUnit.MILLISECONDS);
+            return(req.process(getSolrClient()).getResponse().get("uniqueKey").toString());
+        } catch (SolrServerException | IOException e) {
+            getLogger().error("Solr query to retrieve uniqueKey-field failed due to {}", new Object[]{solrQuery.toString(), e}, e);
+            throw new ProcessException(e);
+        }
+    }
 
-            final SolrDocumentList documentList = response.getResults();
-            logger.info("Retrieved {} results from Solr for {} in {} ms",
-                    new Object[] {documentList.getNumFound(), query, duration});
+    @Override
+    public void onTrigger(final ProcessContext context, final ProcessSession session) throws ProcessException {
 
-            if (documentList != null && documentList.getNumFound() > 0) {
-                FlowFile flowFile = session.create();
-                flowFile = session.write(flowFile, new QueryResponseOutputStreamCallback(response));
-                flowFile = session.putAttribute(flowFile, CoreAttributes.MIME_TYPE.key(), "application/xml");
-                session.transfer(flowFile, REL_SUCCESS);
+        final ComponentLog logger = getLogger();
+        final AtomicBoolean continuePaging = new AtomicBoolean(true);
+        final SolrQuery solrQuery = new SolrQuery();
 
-                StringBuilder transitUri = new StringBuilder("solr://");
-                transitUri.append(getSolrLocation());
-                if (SOLR_TYPE_CLOUD.equals(context.getProperty(SOLR_TYPE).getValue())) {
-                    transitUri.append("/").append(context.getProperty(COLLECTION).getValue());
-                }
+        try {
+            if (id_field == null) {
+                id_field = getFieldNameOfUniqueKey();
+            }
 
-                session.getProvenanceReporter().receive(flowFile, transitUri.toString(), duration);
+            final String dateField = context.getProperty(DATE_FIELD).getValue();
 
-                // if initialized then page through the results and send out each page
-                if (initialized) {
-                    int endRow = response.getResults().size();
-                    long totalResults = response.getResults().getNumFound();
+            solrQuery.setQuery("*:*");
+            final String query = context.getProperty(SOLR_QUERY).getValue();
+            if (!StringUtils.isBlank(query) && !query.equals("*:*")) {
+                solrQuery.addFilterQuery(query);
+            }
+            final StringBuilder automatedFilterQuery = (new StringBuilder())
+                    .append(dateField)
+                    .append(":[")
+                    .append(context.getStateManager().getState(Scope.CLUSTER).get(STATE_MANAGER_FILTER))
```

Comment:
There are couple of `context.getStateManager().getState(Scope.CLUSTER)` call. This should be consolidated by having a variable holding the result of this method call, to avoid accessing Zookeeper often.